src/test/test_event.py

```diff
@@ -0,0 +1,241 @@
+
+    def setUp(self) -> None:
+        pass
+
+    def tearDown(self) -> None:
+        pass
+
+    def test_create_event(self):
+        event_option = EventOptions(user_id="test_user_id")
+        self.assertTrue(isinstance(event_option, EventOptions))
+        self.assertFalse(isinstance(event_option, BaseEvent))
+        self.assertTrue("user_id" in event_option)
+        self.assertFalse("device_id" in event_option)
+        self.assertEqual("test_user_id", event_option.user_id)
+        self.assertEqual("test_user_id", event_option["user_id"])
+        self.assertEqual(0, event_option.retry)
+        event_option.retry += 1
+        self.assertEqual(1, event_option.retry)
+        with self.assertLogs(None, "ERROR") as cm:
+            event_option["id_device"] = "test_device_id"
+            event_option["time"] = 98.6
+            self.assertFalse("id_device" in event_option)
+            self.assertFalse("time" in event_option)
+            self.assertFalse("device_id" in event_option)
+            self.assertIsNone(event_option["device_id"])
+            self.assertIsNone(event_option["id_device"])
+            self.assertIsNone(event_option["time"])
+            self.assertEqual(["ERROR:amplitude.event:Unexpected event property key: id_device",
+                              "ERROR:amplitude.event:Event property time value type: <class 'float'>. Expect <class 'int'>"],
+                             cm.output)
+        event_option.event_id = 10
+        self.assertEqual('{"event_id": 10, "user_id": "test_user_id"}', str(event_option))
+        event_option["plan"] = Plan(branch="test_branch", version_id="v1.1")
+        self.assertEqual({"user_id": "test_user_id",
+                          "event_id": 10,
+                          "plan": {"branch": "test_branch", "versionId": "v1.1"}}, event_option.get_event_body())
+        event = BaseEvent(event_type="test_event", event_properties={"properties1": "test"})
+        self.assertTrue(isinstance(event, EventOptions))
+        self.assertTrue(isinstance(event, BaseEvent))
+        self.assertFalse(isinstance(event, IdentifyEvent))
+        self.assertFalse(isinstance(event, GroupIdentifyEvent))
+        self.assertFalse(isinstance(event, RevenueEvent))
+        self.assertFalse("user_id" in event)
+        self.assertFalse("plan" in event)
+        self.assertFalse("device_id" in event)
+        event.load_event_options(None)
+        self.assertFalse("user_id" in event)
+        self.assertFalse("plan" in event)
+        self.assertFalse("device_id" in event)
+        event.load_event_options(event_option)
+        self.assertTrue("user_id" in event)
+        self.assertTrue("plan" in event)
+        self.assertFalse("device_id" in event)
+        identify_event = IdentifyEvent()
+        self.assertTrue(isinstance(identify_event, EventOptions))
+        self.assertTrue(isinstance(identify_event, BaseEvent))
+        self.assertTrue(isinstance(identify_event, IdentifyEvent))
```

Comment:
I guess it won't hurt to verify the inheritance, maybe not that necessary though.